proposed/2022/DotnetListPackageMachineReadableJsonOutput.md

```diff
@@ -772,6 +772,172 @@ Outputs json for format version 1, if it's not specified then latest version'll
 }
 ```
 
+### Error handling
+
+In case of an error it would be written into `problems` section and return non-zero error exit code to indicate there had been some error to help scripting use cases.
+For current implementation we'll returns general error code `1` from application, more specific error code for each error scenario is not considered in this spec scope.
+In case there is parameter, options error then it may defaults back to console output instead of json output.
```

Comment:
Can we do a list of things that we know would error out?

Reply:
I'm still discovering those, I feel it's more of implementation details, before listing them out we need to agree on whether we error out on console and exit or include errors in `problems` section of json?

Reply:
It was a formatting suggestion, not a content one.

You have sentences enumerating those failure cases. Making them a list would be more readable

Reply:
Rereading it, I'm not sure I fully understand what these sentences are trying to say. 


When there's an error, the error details will be written into a `problems` section and a non-0 error code will be returned. 

In case there bad parameter input, such as options error the json might not be returned?
Do we have an error and no json output generated?

Reply:
For example, `dotnet list package -include-transitive --format json`, here `-include-transitive` missing additional `-` in front. But parameter validation happens way before json rendering, so we couldn't return json result even though customer requested one.

```
Specify --help for a list of available options and commands.
error: Unrecognized option '-include-transitive'


Usage: NuGet.CommandLine.XPlat.dll package list [arguments] [options]

Arguments:
  <PROJECT | SOLUTION>  A path to a project, solution file or directory.
```

Reply:
I see. Ok that makes sense.

What about dotnet list package --framework where the framework doesn't exist?

Reply:
> I see. Ok that makes sense.
> 
> What about dotnet list package --framework where the framework doesn't exist?

It wouldn't fail before generating json, it'll return empty json. I manually tested with `net7.0` framework option.

Reply:
ok. I see. 

I think those are the types of scenarios that'd be great to enumerate. 

It doesn't have to be a complete list, but largely what we discussed, something to give folks a good sense about what happens.

Reply:
Ok, I added above known scenarios. We can update later even this PR is merged.

Reply:
Just having 2nd thought about printing json output on console. Since it's machine-readable output instead of trying to write json to console, just writing to json file would have been cleaner and display all the error and message as console window does as now. Then it would be more consistent, expected experience for end users for both console and json output scenario.